proxy_connect.go

```diff
@@ -76,16 +77,15 @@ func (proxy *proxy) nextCONNECT(downstream net.Conn) filters.Next {
 			return resp, nextCtx, nil
 		}
 
+		_, span := trace.StartSpan(ctx, fmt.Sprintf("connect dial:%v", modifiedReq.Host))
```

Comment:
Something that would be nice is rather than introducing a direct dependency on opencensus, just provide generic tracing hooks in here that allow consumers of the proxy library to plug in their own tracing via the opts structure. Aside from being more flexible, it also makes it easy to unit test that tracing by configuring proxy with a test tracer.

Reply:
That's really the main point of opencensus -- it's a generic tracing API with many different implementations, including ones we could write if we wanted to.

Reply:
I.e. as written for the `proxy` repo proper, this is a no-op. The lantern client or http proxy, or any other user of `proxy` needs to set the tracer.